source/Nevermore/Util/DataModificationQueryBuilder.cs

```diff
@@ -34,20 +38,21 @@ public DataModificationQueryBuilder(IRelationalStoreConfiguration configuration,
             this.keyAllocator = keyAllocator;
         }
 
-        public PreparedCommand PrepareInsert(IReadOnlyList<object> documents, InsertOptions options = null)
+        public PreparedCommand[] PrepareInsert(IReadOnlyList<object> documents, InsertOptions options = null)
         {
+            var commands = new List<PreparedCommand>();
```

Comment:
Do we need this list here? we're only adding to it once

Reply:
Nice pickup, we don't need it at all actually as all the commands are now created in 1 place.